### PR TITLE
feat(github): add release action

### LINKED ISSUE #89: write a github action to upload compiled assets
To prepare for the 0.1.0 release, we should set up a post-merge github action to produce a package containing krustlet, the README and the LICENSE file.

https://help.github.com/en/actions/publishing-packages-with-github-actions/about-packaging-with-github-actions

https://help.github.com/en/packages/publishing-and-managing-packages/about-github-packages

Or we can use my github action and upload the packages to WABS: https://github.com/marketplace/actions/azure-blob-storage-upload

### PR DESCRIPTION
**Pull request by bacongobbler**
This action workflow is triggered on a new tagged release. When released, this action builds the release target for krustlet-wasi and krustlet-wascc, packages them up in a tarball, generates a checksum, uploads the crate to crates.io, and uploads the release assets to Azure Blob Storage.

TODO:

- [x] set up `secrets.CargoToken`
- [x] set up `secrets.AzureStorageConnectionString`

closes #89

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          cd _dist
+          sha256sum *.tar.gz > checksums.txt
+
+      - name: publish to crates.io
+        uses: actions-rs/cargo@v1
+        with:
+          command: publish
+          args: --token ${{ secrets.CargoToken }}
+
+      - name: upload tarballs to WABS
```
Comment (radu-matei) [inline .github/workflows/release.yml]: nit: I assume `WABS` is Windows Azure Blob Storage? 
(I don't think that's the name anymore, and could we avoid the acronym?)

Comment (bacongobbler) [reply .github/workflows/release.yml]: absolutely. I just like saying the acronym wabs :)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        with:
+          command: build
+          args: --release --all-features
+
+      - name: package release assets
+        run: |
+          mkdir _dist
+          cp README.md LICENSE target/release/krustlet-wasi target/release/krustlet-wascc _dist/
+          cd _dist
+          tar czf krustlet-${{ steps.get_version.outputs.VERSION }}-linux-amd64.tar.gz README.md LICENSE krustlet-wasi krustlet-wascc
```
Comment (radu-matei) [inline .github/workflows/release.yml]: As we hopefully add support for more platforms, I assume we will update this line to include the said platforms?

Comment (bacongobbler) [reply .github/workflows/release.yml]: correct

Comment (thomastaylor312) [inline .github/workflows/release.yml]: Any reason we can't just add in the macOS build right now?

Comment (bacongobbler) [reply .github/workflows/release.yml]: No particular reason. Ill see what I can do.

Comment (bacongobbler) [reply .github/workflows/release.yml]: done

File: .github/workflows/canary.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        os: [macos-latest, ubuntu-latest]
+        arch: [amd64]
+    steps:
+      - uses: actions/checkout@v2
+      - name: build release
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --all-features
+      - name: package release assets
```
Comment (rylev) [inline .github/workflows/canary.yml]: There seems to be very large overlap between canary and release. Does GitHub actions support templates like Azure DevOps does?

Comment (bacongobbler) [reply .github/workflows/canary.yml]: it should yeah! I'll see what I can do to share code here. There are some subtle differences between the canary and release workflows, but perhaps there are ways to exclude certain jobs (or to check if it's a push event vs a tag event)

Comment (bacongobbler) [reply .github/workflows/canary.yml]: okay, I think I managed to consolidate the two using conditionals.

**Pull request merged**